modules/core/src/main/scala/org/tessellation/Main.scala

```diff
@@ -51,7 +52,8 @@ object Main
       _ <- IO.unit.asResource
       p2pClient = P2PClient.make[IO](sdkP2PClient, sdkResources.client, sdkServices.session)
       queues <- Queues.make[IO](sdkQueues).asResource
-      storages <- Storages.make[IO](sdkStorages, cfg.snapshot).asResource
+      maybeRollbackHash = Option(method).collect { case rr: RunRollback => rr.rollbackHash }
+      storages <- Storages.make[IO](sdkStorages, cfg.snapshot, maybeRollbackHash).asResource
```

Comment:
How does it differ from the following code that is already used for running genesis?
```
storages.globalSnapshot.prepend(...)
```

Reply:
Prepend will try to save it to disk again hence an error will be thrown constantly because of retry mechanism (such file already exists). By setting caches manually we don't have that error.